### ISSUE TITLE
Buy token in localhost:7545

### ISSUE THREAD

**Issue by Nghialm** [dappuniversity/token_sale#9]
Hi, I get you solution and run buy tokens in web.
It has error below,
Please check for me
![20token_erro](https://user-images.githubusercontent.com/9886111/44563382-47fdb400-a788-11e8-8272-f24e8c8d2736.png)

Thanks

**Comment by ZainaliSyed** [comment]
@gwmccubbin  I am also facing same issue , please guide us what i am missing 
![screen shot 2018-09-04 at 9 22 12 pm](https://user-images.githubusercontent.com/24393023/45044212-e5aa8a80-b088-11e8-878b-216f7e15f652.png)

**Comment by gwmccubbin** [comment]
@ZainaliSyed @Nghialm Hey there! Have you all tried resetting your Metamask account? You always need to do this after restarting ganache:

https://consensys.zendesk.com/hc/en-us/articles/360004177531-Resetting-an-Account-New-UI-

Also, do you still get this issue of you clone the code from the master branch?

**Comment by ZainaliSyed** [comment]
@gwmccubbin  

Thank you for your time , Still facing same i had , **failure in MetaMask transaction** 

1. git pull origin master
2. reset my MetaMask Account as well
3. truffle compile 
4. truffle migtate --network=development --reset

Meta MAsk Version:  4.9.3
Ganache version : 1.2.1

**Comment by gwmccubbin** [comment]
@ZainaliSyed Try this order?

git pull origin master
truffle compile
truffle migtate --network=development --reset
reset my MetaMask Account as well

**Comment by joaocarlosleme** [comment]
Same problem here. Solved by commenting out `//require(tokenContract.balanceOf(this) >= _numberOfTokens);` on DappTokenSale.sol (tokens are on Admin account, not ContractAccount I guess), and also commenting out `//require(balanceOf[msg.sender] >= _value);` on DappToken.sol, because sender may start with 0 dapps.

@gwmccubbin please check it out and let me know.

**Comment by ahsidd96** [comment]
@joaocarlosleme yes commenting out the required lines do solve the problem but is it really safe  (in terms of coding) to do this?
@gwmccubbin Please help us out.

**Comment by joaocarlosleme** [comment]
> @joaocarlosleme yes commenting out the required lines do solve the problem but is it really safe (in terms of coding) to do this?
> @gwmccubbin Please help us out.

Commenting out those lines was a workaround to build but the actual solution is to transfer tokens from the account creator to the DappTokenSale.address as I mentioned on [this answer on stackexchange](https://ethereum.stackexchange.com/a/65027/20639).
